spec/linode_spec.rb

```diff
@@ -93,6 +93,80 @@
       @linode.api_key.should == @api_key
     end
 
+    it 'should request the API key only when accessed' do
```

Comment:
This will basically catch the case where, through some change in the future, creating a Linode with a username and password fetches the API key immediately, that'd defeat the purpose of `fetch_api_key` being public, as a request would've already been made with the default arguments.